plugins/createitem.cpp

```diff
@@ -107,126 +100,225 @@ bool makeItem (df::unit *unit, df::item_type type, int16_t subtype, int16_t mat_
+    auto item_type = item_type::NONE;
     int16_t item_subtype = -1;
     int16_t mat_type = -1;
     int32_t mat_index = -1;
     int32_t growth_print = -1;
     int count = 1;
     bool move_to_cursor = false;
 
-    if (parameters.size() == 1)
-    {
-        if (parameters[0] == "inspect")
-        {
+    if (parameters.size() == 1) {
+        if (parameters[0] == "inspect") {
             CoreSuspender suspend;
-            df::item *item = Gui::getSelectedItem(out);
+            auto item = Gui::getSelectedItem(out);
             if (!item)
-            {
                 return CR_FAILURE;
-            }
 
             ItemTypeInfo iinfo(item->getType(), item->getSubtype());
             MaterialInfo minfo(item->getMaterial(), item->getMaterialIndex());
             out.print("%s %s\n", iinfo.getToken().c_str(), minfo.getToken().c_str());
             return CR_OK;
         }
-        else if (parameters[0] == "floor")
-        {
+        else if (parameters[0] == "floor") {
             dest_container = -1;
             dest_building = -1;
             out.print("Items created will be placed on the floor.\n");
             return CR_OK;
         }
-        else if (parameters[0] == "item")
-        {
+        else if (parameters[0] == "item") {
             dest_building = -1;
-            df::item *item = Gui::getSelectedItem(out);
-            if (!item)
-            {
+            auto item = Gui::getSelectedItem(out);
+            if (!item) {
                 out.printerr("You must select a container!\n");
                 return CR_FAILURE;
             }
             switch (item->getType())
-            {
-            case item_type::FLASK:
-            case item_type::BARREL:
-            case item_type::BUCKET:
-            case item_type::ANIMALTRAP:
-            case item_type::BOX:
-            case item_type::BAG:
-            case item_type::BIN:
-            case item_type::BACKPACK:
-            case item_type::QUIVER:
-                break;
-            case item_type::TOOL:
-                if (item->hasToolUse(tool_uses::LIQUID_CONTAINER))
-                    break;
-                if (item->hasToolUse(tool_uses::FOOD_STORAGE))
-                    break;
-                if (item->hasToolUse(tool_uses::SMALL_OBJECT_STORAGE))
+            {   using namespace df::enums::item_type;
+                case FLASK:
+                case BARREL:
+                case BUCKET:
+                case ANIMALTRAP:
+                case BOX:
+                case BAG:
+                case BIN:
+                case BACKPACK:
+                case QUIVER:
                     break;
-                if (item->hasToolUse(tool_uses::TRACK_CART))
-                    break;
-            default:
-                out.printerr("The selected item cannot be used for item storage!\n");
-                return CR_FAILURE;
+                case TOOL:
+                    if (item->hasToolUse(tool_uses::LIQUID_CONTAINER) ||
```

Comment:
will this allow liquids to be spawned inside of containers? were we not able to do that before?

Reply:
This was just moved down from above, so the default is easier to find at the end of the switch.